lib/plan-finder.js

```diff
@@ -259,10 +266,56 @@ class PlanFinder {
         if (!this.isMimeType(outputInstructions.type)) {
             throw new RenditionFormatUnsupportedError(`outputInstructions.type (${outputInstructions.type}) is not propertly formatted`);
         }
-        const plan = this.BFS(source, outputInstructions);
+
+        let plan;
+        // get image metadata and check for "Orientation" property
+        const metadata = await this.getImageMetadata(source.path);
+        // if image has Orientation metadata set image to sensei compliant orientation
+        if (metadata.Orientation) {
+            let tempPlan = new Plan();
+            // make instructions for orientation
+            const instructions = {input: source, output: { type: source.type }};
+            tempPlan.add("workerCallback-worker-flite", instructions);
+            plan = this.BFS(source, outputInstructions, tempPlan);
+        }
+        else {
+            plan = this.BFS(source, outputInstructions);
+        }
         return this.formPlan(source, outputInstructions, plan);
     }
 
+    async getImageMetadata(path) {
+        let metadata = {};
+        if (path && (await fs.pathExists(path))) {
+            const command = `exiftool -json "${path}"`;
+
+            try {
+                console.log(`Running command: ${command}`);
+                const stdout = child_process.execSync(command, {
+                    timeout: EXIFTOOL_TIMEOUT_MS
+                });
+                const rawJSON = JSON.parse(stdout)[0];
+                metadata = rawJSON;
+            } catch (error) {
+                console.error(`error getting metadata via exiftool for source file: ${path}`);
+                console.error(error);
+            }
+            return metadata;  // returns default {} if no metadata found or error
+        } else if (typeof (path) === 'string' && validUrl.isHttpsUri(path)) {
+            try {
+                const result = await probe(path);
```

Comment:
`probe-image-size` will only get the resolution (widthxheight) not the orientation. You need a different tool for urls, I suggest trying `curl+exiftool` with a timeout like we do for exiftool:
```
curl -s ${url} | exiftool -
```

also, please separate this code in a different file

Reply:
here is an example of using curl + exiftool: https://git.corp.adobe.com/nui/core/blob/83ea399f78b0e9cb3a5acb71db2011d77ca7ae10/lib/exiftool-metadata.js#L34-L40
( i was playing around with some of the parameters to get it to run quicker or read less bytes)
however, I would just use the simple cmd like I shared above and use a timeout on the child_process